pkg/controllers/state/statenode.go

```diff
@@ -221,6 +222,19 @@ func (in *StateNode) Initialized() bool {
 	return true
 }
 
+// Disruptable returns if a node has been initialized long enough for Karpenter to disrupt it
+func (in *StateNode) Disruptable(clock clock.Clock) bool {
+	if !in.Initialized() {
+		return false
+	}
+	// A node is disruptable only if it's been initialized for 30 seconds.
+	// This is critical to ensure that nodes that are created as part of a replacement aren't
+	// chosen for disruption as soon as they're initialized. We only need a small time frame
+	// to get one pod nomination event, as that will block candidacy further.
+	cond := in.NodeClaim.StatusConditions().GetCondition(v1beta1.Initialized)
```

Comment:
You need to add this check for emptiness. Otherwise, we are going to mark NodeClaims as empty that are also within this buffer

Reply:
This is an interesting point. On one hand, we're expecting nodes to get pods scheduled onto them soon, which could result in a redundant double patch (mark as empty, then mark as not empty). On the other hand, the 30s is a heuristic on how long we're expecting this to be _good enough_ for pod scheduling. Since we're using the initialization buffer to hold on disrupting, I'm not sure I agree with holding on identifying status conditions of individual node claims. It may be preferable to have the emptiness condition progress while the buffer is going.